docs/technical-overview.md

```diff
@@ -0,0 +1,241 @@
+
+
+#### Requests interface
+
+The requests page (`/requests`) is the primary functional screen of the Gateway Frontend.
+It is available to researchers, reviewers and organization coordinators.
+
+It consists of a number of sub-views:
+
+`/my-requests` / `my-organisations` / `my-reviews` — role-dependent Request Overview page; displays existing requests and their statuses.
+
+`/new` / `/edit` — New/Edit Request form; creates draft request items via `PUT api/requests` and submits them for review via `GET api/requests/{uuid}/submit`.
+
+`/detail/{request-guid}` — Request Detail view that displays the current review step and lists attached files.
+
+
+### Gateway Server
+
+Gateway server is a Java/Spring app that serves as the core component of the Podium system.
+It contains entirety of the business-logic for handling Requests and Reviews, including sending emails to involved parties.
+
+Concrete steps of the Review and Delivery processes are defined with Flowable in BPMN 2.0 format.
+
+Gateway server uses ElasticSearch index for Delivery and Review Process info, Request Details and Review Feedback.
+
+PostgreSQL DB is used to store Delivery and Review Process info, Summary Entries, Review Feedback, Principal Investigator info, and Request entries, details, files and templates.
+Liquibase is used to initialize/update PostgreSQL DB automatically.
+
+Spring profile, data sources, caching options and other configuration parameters are defined in yaml files included in project resources.
+
+Current Gateway Server version: 1.0.2; supported JDK version: 8.
+
+
+### Podium Common Package
+
+The podium-common package contains a number of shared methods and constants used by UAA and Gateway components.
+
+It includes shared DTOs, request type and status enums, event definitions, exception classes and various helper methods that are necessary as "plumbing" / glue code, but do not contain any business-logic.
+
+
+## Workflows
+
+Request review workflow:
+```plantuml
+@startuml
+
+(*) --> "Validation"
+"Revision" ->[Resubmit] "Validation"
+"Validation" ->[Invalid] "Closed"
+"Validation" --> "Review"
+"Review" ->[Approve/reject] "Closed"
+"Review" ->[Revise] "Revision"
+"Validation" ->[Revise] "Revision"
+
+@enduml
+```
+
+Delivery workflow:
+
+```plantuml
+@startuml
+
+(*) --> "Preparation"
+"Preparation" ->[Cancel] "Closed"
+"Preparation" ->[Release] "Released"
+"Released" ->[Received/cancelled] "Closed"
+
+@enduml
+```
+
+
+## Extra notes and upgrade suggestions
+
+Certain implementation nuances might result in system malfunction after the upgrade, the most obvious one being incorrect HTTP Verbs usage.
+Since request entities are created via HTTP `GET` requests, this functionality can break even after a simple proxy configuration change as `GET` requests are often cached.
+It would be better to replace every HTTP `GET` endpoint mapping intended to cause side-effects with an appropriate `PUT`/`POST` verb before performing any other code/dependency changes.
```

Comment:
Agree. These are mainly the submit/approve/reject/etc actions, right?

Reply:
Right, e.g. `@GetMapping("/requests/{uuid}/approve")`.